e2e/scenario_helpers_test.go

```diff
@@ -125,7 +125,8 @@ func executeScenario(ctx context.Context, t *testing.T, opts *scenarioRunOpts) {
 	t.Logf("vmss %s creation succeeded, proceeding with node readiness and pod checks...", vmssName)
 	nodeName := validateNodeHealth(ctx, t, opts.clusterConfig.Kube, vmssName)
 
-	if opts.nbc.AgentPoolProfile.WorkloadRuntime == datamodel.WasmWasi {
+	// skip when outbound type is block as the wasm will create pod from gcr, however, network isolated cluster scenario will block egress traffic of gcr.
+	if opts.nbc.AgentPoolProfile.WorkloadRuntime == datamodel.WasmWasi && opts.nbc.OutboundType != datamodel.OutboundTypeBlock {
```

Comment:
Should we also add outbound type None to this conditional since that also means airgap?